kb/2024-07-22/harvester_cluster_shutdown_and_restart.md

```diff
@@ -0,0 +1,665 @@
+---
+title: Shutdown and Restart a Harvester Cluster
+description: Details steps about how to shutdown a Harvester cluster and restart it.
+slug: shutdown_and_restart_a_harvester_cluster
+authors:
+  - name: Jian Wang
+    title: Staff Software Engineer
+    url: https://github.com/w13915984028
+    image_url: https://github.com/w13915984028.png
+tags: [harvester, cluster, shutdown, poweroff, replacing, rancher]
+hide_table_of_contents: false
+---
+
+Scenarios:
+
+1. The Harvester cluster is installed with 3+ nodes.
+
+1. The **Rancher manager/server** is deployed independently. (Hereafter it is mentioned as `Rancher manager`)
+
+1. The Harvester cluster is imported to this Rancher manager and works as a node driver.
+
+1. The Rancher manager deploys a couple of downstream K8s clusters, the machines/nodes of those clusters are backed by Harvester VMs.
+
+1. There are also some traditional VMs deployed on the Harvester cluster, which have no direct connection with the Rancher manager.
+
+You plan to move those Harvester nodes geographically, or to power off the whole cluster for some time, it is essential to shutdown the Harvester cluster and restart later.
+
+:::note
+
+2 3 4 are optional if your Harvester cluster is mainly running as an IaaS component. This instruction covers all the above scenarios.
+
+:::
+
+## General Principle
+
+To safely shutdown a Harvester cluster, you need to follow the roughly reverse order of the cluster installation and the workload deployments.
+
+Those facts need to be taken into account particularly:
+
+1. The common methodology of kubernetes operator/controller is to try things continuously until they meet expectations. When the cluster is shutting down node by node, if you don't stop those workloads in advance, they will try hard until the last node is off. It causes the last few nodes to have heavy CPU/memory/network/storage usage and increases the chance of data corruption.
```

Comment:
Write `Kubernetes` in upper case. There are several places spread across the document you need to adapt as well.